### PR TITLE
WIP: angular1 router

### PR DESCRIPTION
**Pull request by shahata**

### CODE REVIEW

File: modules/angular1_router/lib/facades.es5 [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+var StringMapWrapper = {
+  create: function () {
+    return {};
+  },
+
+  set: function (m, k, v) {
+    return m[k] = v;
+  },
+
+  get: function (m, k) {
```
Comment (gkalpak) [inline modules/angular1_router/lib/facades.es5]: Shouldn't this method use `hasOwnProperty()` ?
Same goes for `contains()` and `isEmpty()`.

**Comment by btford** [comment]
We need to chat about `canActivate` and how to deal with the traceur runtime.

Other than that, we decided to use properties on the controller fn to store annotations, and that we'd convert a template + controller pair into a directive so that we don't have two code paths but can still handle legacy controllers.

We also still have lots to do in terms of ergonomics, but if we use fn properties as the canonical storage location, others can build helpers on top to make it nicer to write route configs for Angular 1.

File: modules/angular1_router/lib/facades.es5 (hunk 1 above)
Comment (btford) [inline modules/angular1_router/lib/facades.es5]: It doesn't really matter in this case; you'll only ever have `Object`s passed in created by the router itself, so there won't be anything on the prototype unless the user mucks around with `Object.prototype`.

This file mirrors the implementation of `facade/collection.ts`.

**Comment by btford** [comment]
I'm closing this in favor of #3698. Going to finally get this in today. :smile:

**Pull request closed without merging**